lib/config/generate.js

```diff
@@ -0,0 +1,169 @@
+var Checker = require('../checker');
+var utils = require('../utils');
+var path = require('path');
+var Vow = require('vow');
+var vowNode = require('vow-node');
+var Table = require('cli-table');
+var prompt = require('prompt');
+var get = vowNode.promisify(prompt.get.bind(prompt));
+
+/**
+ * Script that walks the user through the autoconfig flow
+ * @type {String[]}
+ */
+var prompts = [
+    'Please choose a preset number:',
+    'Create an (e)xception for this rule, or (f)ix this issue:'
+];
+
+/**
+ * Generated configuration through the autoconfig flow
+ * @type {Object}
+ */
+var config = {};
+
+/**
+ * Generates a configuration object based for the given path
+ * based on the best fitting preset
+ *
+ * @param  {Object} options
+ * @param  {Object} options.presets - List of registered presets
+ * @param  {Object} options.path - The path containing files used to infer the configuration
+ * @param  {Object} options.checker
+ *
+ * @return {Object} The generated JSCS configuration
+ */
+module.exports = function(options) {
+    options = options || {};
+
+    var checker = options.checker;
+    var _path = options.path;
+    var presetNames = Object.keys(checker.getConfiguration().getPresets());
+
+    if (utils.isRelativePath(_path)) {
+        _path = path.resolve(checker.getConfiguration().getBasePath(), _path);
+    }
+
+    console.log('Checking', _path, 'against the presets');
+
+    prompt.start();
+    prompt.message = '';
+    prompt.delimiter = '';
+
+    return Vow.all(presetNames.map(checkAgainstPreset.bind(null, _path)))
+    .then(function(results) {
+        var errorsCollection = [].concat.apply([], results);
+        var errorCounts = getErrorCounts(errorsCollection);
+
+        outputPresetErrorCounts(presetNames, errorCounts);
```

Comment:
Only output unique errors to avoid count discrepancy between the table and the number of rules the user is walked through for fixup/exception.